### PR TITLE
chore: Add priorities to Jira issues

### PR DESCRIPTION
**Pull request by croyzor**

### CODE REVIEW

**Comment by croyzor** [comment]
This is by far the most involved thing I've done with a github action. @cqc-alec, any clues on how to test this?

**Comment by cqc-alec** [comment]
> This is by far the most involved thing I've done with a github action. @cqc-alec, any clues on how to test this?

I can't think of a practical way to test this except (a) merge it, create an issue, see if it worked; or (b) make a test repo and a test Jira board and do the same thing there. I'd suggest (a); it's not the end of the world if it doesn't work first time.

File: .github/workflows/issue.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 name: New issue
 
 on:
+  pull_request:
+    branches:
+      - main
```
Comment (aborgna-q) [inline .github/workflows/issue.yml:6]: Do we want to create tickets for PRs?

Comment (croyzor) [reply .github/workflows/issue.yml:6]: This is just a temporary change for ease of testing

**Comment by croyzor** [comment]
I've done a little bit of sanity checking, but now I think the way to proceed is to use this in anger and see if it works

File: .github/workflows/issue.yml [hunk 2]
```diff
@@ -24,12 +24,30 @@ jobs:
...
         description: ${{ github.event.issue.html_url }}
         fields: '{"labels": ["hugr"]}'
+    - name: set_priority_var
+      env:
+        GH_PRIORITY: ${{ github.event.issue.label }}
+      run: |
+        MY_RESULT=""
+        case ${GH_PRIORITY} in
+          P-low)
+            MY_RESULT=echo Low;;
```
Comment (aborgna-q) [inline .github/workflows/issue.yml:34]:
```suggestion
            MY_RESULT=Low;;
```

File: .github/workflows/issue.yml [hunk 3]
```diff
@@ -24,12 +24,30 @@ jobs:
...
+          P-medium)
+            MY_RESULT=Medium;;
+          P-high)
+            MY_RESULT=High;;
+          P-critical)
+            MY_RESULT=Highest;;
+        esac
+        echo $MY_RESULT >> $GITHUB_OUTPUT
```
Comment (aborgna-q) [inline .github/workflows/issue.yml:42]: You have to indicate the output variable
```suggestion
        echo "JIRA_PRIORITY=$MY_RESULT" >> $GITHUB_OUTPUT
```

File: .github/workflows/issue.yml [hunk 4]
```diff
@@ -24,12 +24,30 @@ jobs:
         summary: « [hugr] ${{ github.event.issue.title }}»
         description: ${{ github.event.issue.html_url }}
         fields: '{"labels": ["hugr"]}'
+    - name: set_priority_var
+      env:
+        GH_PRIORITY: ${{ github.event.issue.label }}
```
Comment (aborgna-q) [inline .github/workflows/issue.yml:29]: I think there's only a "labels" array?
https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=opened#issues

File: .github/workflows/issue.yml [hunk 5]
```diff
@@ -24,12 +24,30 @@ jobs:
         summary: « [hugr] ${{ github.event.issue.title }}»
         description: ${{ github.event.issue.html_url }}
         fields: '{"labels": ["hugr"]}'
+    - name: set_priority_var
```
Comment (aborgna-q) [inline .github/workflows/issue.yml:27]:
```suggestion
    - name: Parse the priority label into a Jira priority
      id: set_priority_var
```

File: .github/workflows/issue.yml [hunk 6]
```diff
@@ -24,12 +24,30 @@ jobs:
...
+          P-medium)
+            MY_RESULT=Medium;;
+          P-high)
+            MY_RESULT=High;;
+          P-critical)
+            MY_RESULT=Highest;;
+        esac
+        echo $MY_RESULT >> $GITHUB_OUTPUT
+
     - name: Create Task
```
Comment (aborgna-q) [inline .github/workflows/issue.yml:44]: Here, you dropped this
```suggestion
    - name: Create Task
      uses: atlassian/gajira-create@v3.0.1
```

File: .github/workflows/issue.yml [hunk 7]
```diff
@@ -24,12 +24,30 @@ jobs:
...
-      uses: atlassian/gajira-create@v3.0.1
+      env:
+        JIRA_PRIORITY: ${{ steps.set_priority_var.outputs.JIRA_PRIORITY }}
       if: "! contains(github.event.issue.labels.*.name, 'bug')"
```
Comment (aborgna-q) [inline .github/workflows/issue.yml:47]: This works as-is, but the linter works better with `{{`s.
```suggestion
      if: ${{! contains(github.event.issue.labels.*.name, 'bug')}}
```
(you may want to change it on the other `if contains` too)

**Review by aborgna-q — COMMENTED** [review]
The [github actions](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-github-actions) vscode plugin is quite useful for catching errors.

File: .github/workflows/issue.yml (hunk 5 above)
Comment (croyzor) [reply .github/workflows/issue.yml]: Nice, I didn't know you could use a seperate `id`

File: .github/workflows/issue.yml (hunk 6 above)
Comment (croyzor) [reply .github/workflows/issue.yml:45]: 🙏

**Comment by croyzor** [comment]
> The [github actions](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-github-actions) vscode plugin is quite useful for catching errors.

That's good to know! I was looking for some tooling to help me but didn't find this

**Review by aborgna-q — APPROVED** [review]
That should do it!
We'll need to update the other repos too.

**Comment by croyzor** [comment]
I've just added an issue on tket2 to copy this once it works - which other repos do we have jira integration for?

**Pull request merged**